packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -692,16 +730,23 @@ export function createElement(type, config, children) {
       defineKeyPropWarningGetter(props, displayName);
     }
   }
-
+  const trackActualOwner =
+    ReactSharedInternals.recentlyCreatedOwnerStacks++ < ownerStackLimit;
```

Comment:
It's weird to me that the owner stack value below is gated by `__DEV__` but this line is not?

Reply:
It'll be DCE'd